### PR TITLE
[regression](cluster key) add some cluster key case like mow

### PR DESCRIPTION
**Pull request by mymeiyi**

### CODE REVIEW

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Review by Yukang-Lian — APPROVED** [review]

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by mymeiyi** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/master/ReportHandler.java [hunk 1]
```diff
@@ -962,10 +963,19 @@ private static void deleteFromMeta(ListMultimap<Long, Long> tabletDeleteFromMeta
...
+                                    if (indexId == olapTable.getBaseIndexId() || olapTable.isShadowIndex(indexId)) {
+                                        List<Integer> clusterKeyIndexes = OlapTable.getClusterKeyIndexes(
+                                                indexMeta.getSchema());
+                                        if (!CollectionUtils.isEmpty(clusterKeyIndexes)) {
+                                            createReplicaTask.setClusterKeyIndexes(clusterKeyIndexes);
+                                            LOG.info("table: {}, partition: {}, index: {}, tablet: {}, "
+                                                            + "cluster key indexes: {}", tableId, partitionId, indexId,
+                                                    tabletId, clusterKeyIndexes);
+                                        }
+                                    }
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/master/ReportHandler.java:978]: May be we should pass olaptable as arg to createReplicaTask and handle clusterkeyindex in createReplicaTask internally.

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java [hunk 2]
```diff
@@ -2650,6 +2657,10 @@ private boolean createOlapTable(Database db, CreateTableStmt stmt) throws UserEx
             } catch (AnalysisException e) {
                 throw new DdlException(e.getMessage());
             }
+            if (enableUniqueKeyMergeOnWrite && !enableLightSchemaChange && !CollectionUtils.isEmpty(
+                    keysDesc.getClusterKeysColumnNames())) {
+                throw new DdlException("Unique merge-on-write table with cluster keys must enable light schema change");
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:2662]: Unique merge-on-write tables with cluster keys require light schema change to be enabled.

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run cloud_p0

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**